src/ConnectCore.js

```diff
@@ -107,6 +133,18 @@ class ConnectCore {
   }
 
   /**
+   *  Subscribe to different events
+   */
+
+   subscribe(topic) {
```

Comment:
While pub sub is cool. This seems a lot of complexity and a new area for devs to learn. In particular in this use case where it is a one off that we're waiting for.

Ideally a single callback can be used or a function returning a promise. But again in a Button wrapper code and not in ConnectCore